backend/dashboard/tests.py

```diff
@@ -1,3 +1,243 @@
+                                   'customerID': 52, 'amount': 860.27, 'ip_address': '179.152.194.186'}, 
+                             '1': {'action': 'declined', 'time_completed': '2023-12-07T08:07:20.451Z', 
+                                   'longitude': -1.121183, 'merchant_name': 'Smith Corp', 'latitude': 11.962175, 
+                                   'customerID': 74, 'amount': 144.53, 'ip_address': '173.64.65.25'}}
+    def test_get_closest_match(self):
+        # Test matching "3M Co" to "3M Co" in ESG_scores
+        result = get_closest_match("3M", self.esg)
+        self.assertEqual(result, "3M Co")
+        
+        # Test matching "Smith Corp" to "A O Smith Corp"
+        result = get_closest_match("Smith Corp", self.esg)
+        self.assertEqual(result, "A O Smith Corp")
+        
+        # Test when there's no match
+        result = get_closest_match("Uber", self.esg)
+        self.assertEqual(result, None)
+        
+    def test_get_company_env_score(self):
+        transaction = {"merchant_name": "3M Co"}
+        result = get_company_env_score(transaction, self.esg)
+        self.assertEqual(result, 526)
+
+        transaction = {"merchant_name": "A O Smith Corp"}
+        result = get_company_env_score(transaction, self.esg)
+        self.assertEqual(result, 510)
+        
+        # Test matching with "Uber" which should have no match
+        transaction = {"merchant_name": "Uber"}
+        result = get_company_env_score(transaction, self.esg)
+        self.assertEqual(result, 0)
+    
+    
+    def test_get_company_tier(self):
+        self.assertEqual(_company_tier(600), 1)
+        self.assertEqual(_company_tier(501), 2)
+        
+    def test_count_green_transactions_in_period(self):
+        start_date = datetime(2023, 1, 1)
+        end_date = datetime(2024, 1, 1)
+        result = _count_green_transactions_in_period(self.transactions, start_date, end_date, self.esg)
+        self.assertEqual(result, 1)
+        
+    def test_get_unique_companies(self):
+        result = _get_unique_companies(self.transactions, self.esg)
+        self.assertEqual(result, {"3M Co", "A O Smith Corp"})
+    
+    def test_calculate_score(self):
+        start_date = datetime(2023, 1, 1)
+        end_date = datetime(2024, 1, 1)
+        result = calculate_score(self.transactions, start_date, end_date, self.esg)
+        total_spending = 0
+        for transaction in self.transactions:
+            company_env_score = _get_company_env_score(transaction, self.esg)
+            transaction_amount = self.transactions[transaction]['amount']
+            env_contribution += company_env_score * transaction_amount
+            total_spending += transaction_amount
+        expected_result = int(env_contribution / total_spending) if total_spending != 0 else None
+        self.assertEqual(result, expected_result)
+    
+    def test_calculate_company_esg_scores(self):
```

Comment:
The test I was referring to was this test. I thought that the data format was a dictionary of dictionaries.